### ISSUE TITLE
Server.init fails when using Postgres and nest ConfigModule

### ISSUE THREAD

**Issue by ttournie** [vendure-ecommerce/vendure#1345]
**Describe the bug**
When using PostgresInitializer and ConfigModule in a plugin, the server.init fails and we get the following error : 'Cannot destructure property 'superadminCredentials' of 'configService.authOptions' as it is undefined.'

**To Reproduce**
- on a fresh Vendure install create a plugin : 

```
import { Inject } from '@nestjs/common'
import { PluginCommonModule, VendurePlugin, ConfigService } from '@vendure/core'

import { applicationConfig } from './application.config'
import { ConfigModule, ConfigType } from '@nestjs/config'

@VendurePlugin({
  imports: [PluginCommonModule, ConfigModule.forFeature(applicationConfig)],
})
export class TestPlugin {
  readonly logPrefix = '[Job cleanup]'

  constructor(
    @Inject(applicationConfig.KEY)
    private appConfig: ConfigType<typeof applicationConfig>,
    private configService: ConfigService
  ) {}
}
```

Create the application.config :

```
import { registerAs } from '@nestjs/config'

export const applicationConfig = registerAs('app', () => ({
  executionContext: process.env.CONTEXT || 'SERVER',
  nodeEnv: process.env.NODE_ENV,
}))
```

Then create a 2e2 test : 

```
import {
  createTestEnvironment,
  registerInitializer,
  SqljsInitializer,
  PostgresInitializer,
} from '@vendure/testing'
import path from 'path'
import { testConfig } from './config/test-config'
import { initialData } from './config/initial-data'
import { TestPlugin } from '../../../src/plugins/test/test-plugin'

const sqliteDataDir = path.join(__dirname, '__data__');
//registerInitializer('sqljs', new SqljsInitializer(sqliteDataDir));
registerInitializer('postgres', new PostgresInitializer())


describe('products plugin', () => {
  const { server } = createTestEnvironment({
    ...testConfig,
    plugins: [TestPlugin],
  })

  // The timeout needs to be a little long to be able to init the database
  beforeAll(async () => {
    await server.init({
      initialData,
      productsCsvPath: path.join(__dirname, 'config/test-products.csv'),
      customerCount: 1,
    })
  }, 120000)

  afterAll(async () => {
    await server.destroy()
  })

  it('Test', async () => {
    expect(true).toEqual(true)
  })
})
```

Then run the test with Jest

**Expected behavior**
Server.init should work

**Environment (please complete the following information):**

- Dependencies : 
```
    "@nestjs/config": "^1.1.6",
    "@types/jest": "^26.0.20",
    "@vendure/admin-ui-plugin": "1.4.4",
    "@vendure/asset-server-plugin": "1.4.4",
    "@vendure/core": "1.4.4",
    "@vendure/email-plugin": "1.4.4",
    "@vendure/testing": "1.4.3",
    "@vendure/ui-devkit": "1.4.3",
    "jest": "^26.6.3",
    "jest-junit": "^13.0.0",
    "pg": "^8.7.1",
    "sql.js": "1.6.2",
    "ts-jest": "^26.4.4",
    "typescript": "4.3.5"
```

 - Database : postgres

**Comment by michaelbromley** [comment]
I've just been digging in to this. First of all, I think the problem is independent of which DB is being tested against, but it shows up with the postgres driver because the error is being thrown during population of the DB, and in sqlite it is cached so that step is being skipped.

The root cause is that on this line: https://github.com/vendure-ecommerce/vendure/blob/eefd659eaf09a3af6717b7c493d5f6ab5267509a/packages/testing/src/utils/get-superadmin-context.ts#L14
the Nest ConfigModule's `ConfigService` is being returned by the DI container, rather than Vendure's `ConfigService`. 

I'm still trying to figure out why that is but I traced it down to [this code](https://github.com/nestjs/nest/blob/3dca681a086f0c45d5b09bc34db01d5e5cb09aba/packages/core/injector/instance-links-host.ts#L26-L29) inside NestJS: 
```ts
get<T = any>(token: InstanceToken, moduleId?: string): InstanceLink<T> {
    const name = isFunction(token)
      ? (token as Function).name
      : (token as string | symbol);
    const modulesMap = this.instanceLinks.get(name);
```
At this point, the `ConfigService.name` is equal to the string `"ConfigService"`, so this is where it is getting mixed up.

Note that this seems to be fixed in Nest v8+ (Vendure is currently still on v7). But I will see if I can find a work-around for now.

**Comment by michaelbromley** [comment]
OK here's a work-around you can try:

The idea is to solve the problem of 2 providers with identical names. Since Nest is querying `Function.name`, what if we just change the name of one of them? Well, the `Function.name` property is read-only, but we can override that like this:

```ts
Object.defineProperty(ConfigService, 'name', { writable: true });
(ConfigService as any).name = 'VendureConfigService';
```
If you execute that code as early as possible in your tests it might solve it.

Also note that this issue will not just affect your e2e tests, but also any runtime code that is calling `injector.get(ConfigService)` and expective Vendure's ConfigService. So the same work-around would need to be applied to your app code too.

The ultimate solution should come with an upgrade to Nest v8, but this must wait for Vendure v2.0 because it involves breaking changes.

**Comment by ttournie** [comment]
Ok, I see
The workaround seems to work on this exemple. Unfortunatly we still have an issue with this other msg : 
`TypeError: Cannot read property 'value' of undefined` on the `ChannelService.getDefaultChannel`.

I thought it was linked but apparently not so I need to find what cause this other error...

**Comment by ttournie** [comment]
Ok so the `TypeError: Cannot read property 'value' of undefined` was an issue in our end, we declared the ChannelService has a provider in one of our plugin for no reason

**Comment by michaelbromley** [comment]
OK, I will close this now then. The underlying issue for the ConfigModule will be automatically resolved when we upgrade NestJS in v2.0
